iavl_tree_test.go

```diff
@@ -95,6 +96,44 @@ func TestVersionedRandomTreeSmallKeys(t *testing.T) {
 	}
 }
 
+func TestVersionedRandomTreeSmallKeysRandomDeletes(t *testing.T) {
+	require := require.New(t)
+	tree := NewVersionedTree(100, db.NewMemDB())
+	singleVersionTree := NewVersionedTree(0, db.NewMemDB())
+	versions := 30
+	keysPerVersion := 50
+
+	for i := 1; i <= versions; i++ {
+		for j := 0; j < keysPerVersion; j++ {
+			// Keys of size one are likely to be overwritten.
+			k := []byte(cmn.RandStr(1))
+			v := []byte(cmn.RandStr(8))
+			tree.Set(k, v)
+			singleVersionTree.Set(k, v)
+		}
+		tree.SaveVersion(uint64(i))
+	}
+	singleVersionTree.SaveVersion(1)
+
+	for _, i := range rand.Perm(versions) {
```

Comment:
This will cover [1, versions]
The non-random test deleted [1, versions)

I think you must do `rand.Perm(versions-1)`

Reply:
Yeah I'm aware, but the delete of `version` will just silently fail so shouldn't matter? Or am I missing something?

Reply:
Fixed anyway, it is more correct.